### PR TITLE
transmission: use procd

### PR DESCRIPTION
**Pull request by nmav**
That allows to restart transmission when it crashes, to limit the memory used by it, as well as be jailed in the directories it is supposed to access.

Signed-off-by: Nikos Mavrogiannopoulos <nmav@gnutls.org>

### CODE REVIEW

**Comment by nmav** [comment]
I'd appreciate more testing especially in the new jails part.

**Comment by nmav** [comment]
@obsy

File: net/transmission/files/transmission.init [hunk 1]
```diff
@@ -40,14 +42,24 @@ section_enabled() {
 	[ $enabled -gt 0 ]
 }
 
-start_instance() {
-	local s="$1"
+transmission() {
+	local cfg="$1"
+	#give transmission 3/5 of the memory of the system
+	local USE
+	local MEM=`cat /proc/meminfo |grep MemTotal|sed 's/ \+/ /g'|cut -d ' ' -f 2`
```
Comment (karlp) [inline net/transmission/files/transmission.init]: grep MemTotal /proc/meminfo cuts out one shell invocation.

File: net/transmission/files/transmission.init [hunk 2]
```diff
@@ -40,14 +42,24 @@ section_enabled() {
...
+	if test "$MEM" -gt 1;then
+		USE=`expr $MEM \* 3000`
+		USE=`expr $USE / 5`
```
Comment (karlp) [inline net/transmission/files/transmission.init]: You can do this on one line you know. Less shells again.

File: net/transmission/files/transmission.init [hunk 3]
```diff
@@ -40,14 +42,24 @@ section_enabled() {
 	[ $enabled -gt 0 ]
 }
 
-start_instance() {
-	local s="$1"
+transmission() {
+	local cfg="$1"
+	#give transmission 3/5 of the memory of the system
```
Comment (karlp) [inline net/transmission/files/transmission.init]: This isn't configurable?

**Comment by karlp** [comment]
Further $() vs `` is both nestable, and posix, `` is deprecated

**Comment by nmav** [comment]
Karl, the new version should address the issues you raised.

**Comment by dangowrt** [comment]
@nmav: The memory detection didn't work for me, I had to change it to:
```
local MEM=$( cat /proc/meminfo | grep MemTotal | sed 's/ \+/ /g' | cut -d':' -f2 | cut -d'k' -f1 | xargs )
```
This trims the resulting string and removes unwanted whitespace (which is also why using ' ' as field seperator failed for me).

**Comment by dangowrt** [comment]
handling of sub-sequent white-space in `cut` seems to differ when using musl instead of uClibc... So on musl `grep MemTotal /proc/meminfo|sed 's/ \+/ /g'|cut -d ' ' -f 2` doesn't work.
`grep MemTotal /proc/meminfo|sed 's/ \+/ /g'|cut -d ':' -f 2|cut -d 'k' -f 1|xargs`
works on both, uClibc/glibc and musl.

**Comment by jow-** [comment]
And instead of doing five shell invocations to extrac a number I suggest using `sed -ne 's!^MemTotal:[[:space:]]*\([0-9]*\) kB$!\1!p' /proc/meminfo` - thats just one and it should be portable accross all libc implementations :)

**Comment by dangowrt** [comment]
@jow- nice one! I don't know why `sed 's/ \+/ /g'` was part of the original cut-pipery, I haven't seen any `+` characters in meminfo output. maybe @nmav knows more?

**Comment by nmav** [comment]
The 's/ \+/ /g' combines multiple spaces into a single. I don't like much the dependence on the kB, but I guess we have to depend on it anyway or the calculations will be wrong.

**Comment by dangowrt** [comment]
Ah, regex, sure... ` +` reads as one-or-more space... So that's what doesn't work well with busybox-sed on musl...

**Comment by obsy** [comment]
Acked-by: Cezary Jackiewicz <cezary.jackiewicz@gmail.com>

**Comment by nmav** [comment]
Thanks

**Pull request merged**